test/unit/preprocessing/informationfilter/test_sklearn_compat.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import numpy as np
+import pandas as pd
+from sklearn.utils import estimator_checks
+
+from fairlearn.preprocessing import InformationFilter
+
+
+@pytest.mark.parametrize(
+    "test_fn",
+    [
+        # transformer checks
+        estimator_checks.check_transformer_data_not_an_array,
+        estimator_checks.check_transformer_general,
+        estimator_checks.check_transformers_unfitted,
+        # general estimator checks
+        estimator_checks.check_fit2d_predict1d,
+        # estimator_checks.check_methods_subset_invariance,
+        estimator_checks.check_fit2d_1sample,
+        estimator_checks.check_fit2d_1feature,
+        estimator_checks.check_fit1d,
+        estimator_checks.check_get_params_invariance,
+        estimator_checks.check_set_params,
+        estimator_checks.check_dict_unchanged,
+        estimator_checks.check_dont_overwrite_parameters,
+        # nonmeta_checks
+        estimator_checks.check_estimators_dtypes,
+        estimator_checks.check_fit_score_takes_y,
+        estimator_checks.check_dtype_object,
+        estimator_checks.check_sample_weights_pandas_series,
+        estimator_checks.check_sample_weights_list,
+        estimator_checks.check_sample_weights_invariance,
+        estimator_checks.check_estimators_fit_returns_self,
+        estimator_checks.check_complex_data,
+        estimator_checks.check_estimators_empty_data_messages,
+        estimator_checks.check_pipeline_consistency,
+        estimator_checks.check_estimators_nan_inf,
+        estimator_checks.check_estimators_overwrite_params,
+        estimator_checks.check_estimator_sparse_data,
+        estimator_checks.check_estimators_pickle,
+    ]
+)
+def test_estimator_checks(test_fn):
+    test_fn(InformationFilter.__name__, InformationFilter(columns=[]))
+    test_fn(InformationFilter.__name__, InformationFilter(columns=[0]))
+
+
+@pytest.mark.parametrize("transform", [np.array, pd.DataFrame, list])
+def test_linear_dependence(transform):
+    X = np.array([[0, 0, 1, 1, ],
+                  [1, 1, 2, 2, ],
+                  [0.1, 0.2, 1.2, 1.1, ]]).T
+    X_sensitive = X[:, 0]
+    X = transform(X)
+
+    X_tfm = InformationFilter(columns=[0]).fit(X).transform(X)
+
+    cov_col_0 = np.cov(X_sensitive, X_tfm[:, 0])
+    cov_col_1 = np.cov(X_sensitive, X_tfm[:, 1])
+    assert np.isclose(cov_col_0[0, 1], 0.0)
+    assert np.isclose(cov_col_1[0, 1], 0.0)
+    assert X_tfm.shape[1] == 2
```

Comment:
We should have an analogous test when the columns of `X` are name--only for `pd.DataFrame`.

Reply:
Ah this answers my previous question.

Does fairlearn offer an abstraction to check if `X` is a dataframe or an array? It's something we've been munging over on the scikit-lego side of things as well.

Reply:
We have a utility to parametrize the tests to run with all kinds of configurations. You can see that in use here: https://github.com/fairlearn/fairlearn/blob/4dbf97d6fccfbd38be94ac5f7e7f236e4d7ea816/test/unit/reductions/grid_search/test_grid_search_arguments.py#L43

We don't really do anything with X other than passing it to the underlying estimator, so from our perspective we don't even really want to know right now. We do run scikit-learn validation like check_X_y, but we don't take the reformatted output from that.

I'm still not sure I quite understand what you're trying to do. Checking for type is easy with `type(...)` so I must be missing something.